nmosnode/api.py

```diff
@@ -54,19 +55,29 @@ def versionroot(self, api_version):
             abort(404)
         return ["self/","sources/", "flows/", "devices/", "senders/", "receivers/"]
 
-    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/senders/")
-    def senders(self, api_version):
+    @route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/self/")
+    def selfresource(self, api_version):
         if api_version not in NODE_APIVERSIONS:
             abort(404)
-        return self.registry.list_resource("sender", api_version=api_version).values()
+        return self.registry.list_self(api_version=api_version)
+
+    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/<resource_type>/")
+    def resource_list(self, api_version, resource_type):
+        if api_version not in NODE_APIVERSIONS:
+            abort(404)
+        if resource_type not in RESOURCE_TYPES:
+            abort(404)
+        return self.registry.list_resource(resource_type.rstrip("s"), api_version=api_version).values()
 
-    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/receivers/<receiver>/")
-    def receiver_id(self, api_version, receiver):
+    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/<resource_type>/<resource_id>/")
+    def resource_id(self, api_version, resource_type, resource_id):
         if api_version not in NODE_APIVERSIONS:
             abort(404)
-        receivers = self.registry.list_resource("receiver", api_version=api_version)
-        if receiver in receivers:
-            return receivers[receiver]
+        if resource_type not in RESOURCE_TYPES:
+            abort(404)
+        resources = self.registry.list_resource(resource_type.rstrip("s"), api_version=api_version)
+        if resource_id in resources:
```

Comment:
Doesn't need changing, but I'd normally expected to see this done with a try/catch block in Python

Reply:
Yes, I have a bit of a habit of not being fully Pythonic...

Reply:
It's fine, I catch myself doing this sort of thing too but worth bearing in mind ;-)